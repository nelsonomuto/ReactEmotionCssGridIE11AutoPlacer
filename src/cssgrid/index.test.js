import React from 'react';
import renderer from 'react-test-renderer';
import ie11cssGridPrefixer, {
  css,
  AutoPlacer,
  AutoPlace,
  flattenChildren,
  getMediaQueryNodeChildCss,
  unsetStyles
} from './index';

test('getMediaQueryNodeChildCss', () => {
  const node = {
    type: 'MediaQuery',
    props: {
      children: {
        type: 'react.fragment',
        props: {
          children: {
            props: {
              css: css`color: red;`
            }
          }
        }
      }
    }
  }
  const mediaQueryCss = getMediaQueryNodeChildCss(node);
  expect(mediaQueryCss.styles).toEqual('color: red;')
});

describe('unsetStyles ', () => {
  let before = `
    display: some;
    do: that;
    border: 1px solid blue;
    style: this;
  `;
  const after = `
    *, div, p {
      display: unset;
      do: unset;
      border: unset;
      style: unset;
    }
  `.replace(/\s/g,'');
  test('unset function', () => {
    expect(unsetStyles(before).replace(/\s/g,'')).toEqual(after);
  });
  test('cssStyles unset', () => {
    const cssStyles = css`${unsetStyles(before)}`.styles.replace(/\s/g,'');
    expect(cssStyles).toEqual(after);
  });
  test('unset on one line', () => {
    before = `display: some; do: that; border: 1px solid blue; style: this;`;
    expect(unsetStyles(before).replace(/\s/g,'')).toEqual(after);
  });
  test('cssStyles unset on one line', () => {
    const cssStyles = css`${unsetStyles(before)}`.styles.replace(/\s/g,'');
    expect(cssStyles).toEqual(after);
  });
})
describe('flattenChildren', () => {
  test('nested arrays with undefined', () => {
    const items = [
      0,
      undefined,
      [24, [1, 7, undefined]],
      8
    ];
    const flattened = flattenChildren(items);
    console.log('log flattened', { flattened })
    expect(flattened).toEqual([0, 24, 1, 7, 8]);
  });
  test('nested arrays with object', () => {
    const items = [
      0,
      [24, [1, 7, [[500]]]],
      [22, [{j: 5}]],
      8
    ];
    const flattened = flattenChildren(items);
    console.log('log flattened', { flattened })
    expect(flattened).toEqual([0, 24, 1, 7, 500, 22, {j: 5}, 8]);
  });
  test('nested props', () => {
    const items = [
      0,
      {
        type: 'react.fragment',
        props: {
          children: [
            1, 2
          ]
        }
      }
    ];
    const flattened = flattenChildren(items);
    expect(flattened).toEqual([0, 1, 2]);
  });
  test('fragment direct children', () => {
    const items = {
      type: 'react.fragment',
      props: {
        children: [
          0, 1, 2
        ]
      }
    };
    const flattened = flattenChildren(items);
    expect(flattened).toEqual([0, 1, 2]);
  });
  test('nested with props and arrays', () => {
    const items = [
      0,
      {
        type: 'react.fragment',
        props: {
          children: [
            {
              type: 'react.fragment',
              props: {
                children: [1, {
                  type: 'react.fragment',
                  props: {
                    children: [{
                      type: 'react.fragment',
                      props: {
                        children: [2, {
                          type: 'react.fragment',
                          props: {
                            children: { notArray: true, val: 3 }
                          }
                        }],
                      }
                    }]
                  }
                }, [[{ val: 4}]]]
              }
            },
            [5],
            6
          ]
        }
      },
      [7, 8],
      9
    ];
    const flattened = flattenChildren(items);
    expect(flattened).toEqual([0, 1, 2, { notArray: true, val: 3}, { val: 4 }, 5, 6, 7, 8, 9]);
  });
});

test('ie11cssGridPrefixer and css', () => {
  const cols = new Array(7);
  const before = `
      display: grid;
      justify-items: center;
      grid-column-gap: 0.1em;
      grid-row-gap: 0.1em;
      grid-template-columns: repeat(${cols.length}, ${100 / cols.length}%);
      padding: 0px 35px 0 17px;
      position: relative;
      overflow: hidden;
      display: grid;
      grid-template-columns: 100%;
      grid-template-rows: 100%;
    `;
  const after = `
      display: -ms-grid;
      display: grid;
      justify-items: center;
      grid-column-gap: 0.1em;
      grid-row-gap: 0.1em;
      -ms-grid-columns: repeat(7, 14.285714285714286%);
      grid-template-columns: repeat(7, 14.285714285714286%);
      padding: 0px 35px 0 17px;
      position: relative;
      overflow: hidden;
      display: -ms-grid;
      display: grid;
      -ms-grid-columns: 100%;
      grid-template-columns: 100%;
      -ms-grid-rows: 100%;
      grid-template-rows: 100%;
    `;
  expect(ie11cssGridPrefixer(before)).toEqual(after);
  expect(css`${before}`.styles.includes(after)).toEqual(true);
});
test('auto placement', () => {
  const before = `
      thatStyle: visible;
      someStyle: cool;
    `;
  let after = `
      ${before}
      -ms-grid-column: 3;
      -ms-grid-row: 1;
    `;
  const autoPlacer = AutoPlacer({ colLength: 3 });
  autoPlacer.next();
  autoPlacer.next();
  expect(autoPlacer.next(before).styles.trim()).toEqual(after.trim());
  after = `
      ${before}
      -ms-grid-column: 1;
      -ms-grid-row: 2;
    `;
  expect(autoPlacer.next`${before}`.styles.trim()).toEqual(after.trim());
});
test('react component wrapper', () => {
  let component = renderer.create(
    <AutoPlace>
      <p>child one</p>
      <h2>child two</h2>
      <p>child three</p>
      <h3>child four</h3>
      <div>child five</div>
    </AutoPlace>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  component = renderer.create(
    <AutoPlace colLength={3}>
      <p>child one</p>
      <h2>child two</h2>
      <p>child three</p>
      <h3>child four</h3>
      <div>child five</div>
    </AutoPlace>
  );
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  component = renderer.create(
    <AutoPlace colLength={4}>
      <p>child one</p>
      <h2>child two</h2>
      <p>child three</p>
      <h3>child four</h3>
      <div>child five</div>
    </AutoPlace>
  );
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  component = renderer.create(
    <AutoPlace colLength={1}>
      <p>child one</p>
      <h2>child two</h2>
      <p>child three</p>
      <h3>child four</h3>
      <div>child five</div>
    </AutoPlace>
  );
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  component = renderer.create(
    <AutoPlace colLength={5}>
      {new Array(5).fill('').map(() => <p>Test grandChildren 1</p>)}
      {new Array(4).fill('').map(() => <p>Test grandChildren 2</p>)}
    </AutoPlace>
  );
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  component = renderer.create(
    <AutoPlace>
    </AutoPlace>
  );
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  component = renderer.create(
    <AutoPlace>
      <h1>Single</h1>
    </AutoPlace>
  );
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
