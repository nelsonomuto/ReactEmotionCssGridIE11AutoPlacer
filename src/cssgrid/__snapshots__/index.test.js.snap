// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react component wrapper 1`] = `
<div
  className="css-mjwwwb-AutoPlace"
  data-auto-placer-container={true}
  data-children={5}
  data-col-length={5}
>
  <div
    className="css-1tdfgk4"
    data-auto-placer-item={true}
    data-auto-placer-styles="
      -ms-grid-column: 1;
      -ms-grid-row: 1;
    "
    data-has-unset-css-styles={false}
  >
    <p>
      child one
    </p>
  </div>
  <div
    className="css-1pi500h"
    data-auto-placer-item={true}
    data-auto-placer-styles="
      -ms-grid-column: 2;
      -ms-grid-row: 1;
    "
    data-has-unset-css-styles={false}
  >
    <h2>
      child two
    </h2>
  </div>
  <div
    className="css-zzqq5x"
    data-auto-placer-item={true}
    data-auto-placer-styles="
      -ms-grid-column: 3;
      -ms-grid-row: 1;
    "
    data-has-unset-css-styles={false}
  >
    <p>
      child three
    </p>
  </div>
  <div
    className="css-bh2j43"
    data-auto-placer-item={true}
    data-auto-placer-styles="
      -ms-grid-column: 4;
      -ms-grid-row: 1;
    "
    data-has-unset-css-styles={false}
  >
    <h3>
      child four
    </h3>
  </div>
  <div
    className="css-17v5j8a"
    data-auto-placer-item={true}
    data-auto-placer-styles="
      -ms-grid-column: 5;
      -ms-grid-row: 1;
    "
    data-has-unset-css-styles={false}
  >
    <div>
      child five
    </div>
  </div>
</div>
`;

exports[`react component wrapper 2`] = `
<div
  className="css-1emvw2y-AutoPlace"
  colLength={3}
  data-auto-placer-container={true}
  data-children={5}
  data-col-length={3}
>
  <div
    className="css-1tdfgk4"
    data-auto-placer-item={true}
    data-auto-placer-styles="
      -ms-grid-column: 1;
      -ms-grid-row: 1;
    "
    data-has-unset-css-styles={false}
  >
    <p>
      child one
    </p>
  </div>
  <div
    className="css-1pi500h"
    data-auto-placer-item={true}
    data-auto-placer-styles="
      -ms-grid-column: 2;
      -ms-grid-row: 1;
    "
    data-has-unset-css-styles={false}
  >
    <h2>
      child two
    </h2>
  </div>
  <div
    className="css-zzqq5x"
    data-auto-placer-item={true}
    data-auto-placer-styles="
      -ms-grid-column: 3;
      -ms-grid-row: 1;
    "
    data-has-unset-css-styles={false}
  >
    <p>
      child three
    </p>
  </div>
  <div
    className="css-1bmyan6"
    data-auto-placer-item={true}
    data-auto-placer-styles="
      -ms-grid-column: 1;
      -ms-grid-row: 2;
    "
    data-has-unset-css-styles={false}
  >
    <h3>
      child four
    </h3>
  </div>
  <div
    className="css-o4ag6j"
    data-auto-placer-item={true}
    data-auto-placer-styles="
      -ms-grid-column: 2;
      -ms-grid-row: 2;
    "
    data-has-unset-css-styles={false}
  >
    <div>
      child five
    </div>
  </div>
</div>
`;

exports[`react component wrapper 3`] = `
<div
  className="css-1xx69yo-AutoPlace"
  colLength={4}
  data-auto-placer-container={true}
  data-children={5}
  data-col-length={4}
>
  <div
    className="css-1tdfgk4"
    data-auto-placer-item={true}
    data-auto-placer-styles="
      -ms-grid-column: 1;
      -ms-grid-row: 1;
    "
    data-has-unset-css-styles={false}
  >
    <p>
      child one
    </p>
  </div>
  <div
    className="css-1pi500h"
    data-auto-placer-item={true}
    data-auto-placer-styles="
      -ms-grid-column: 2;
      -ms-grid-row: 1;
    "
    data-has-unset-css-styles={false}
  >
    <h2>
      child two
    </h2>
  </div>
  <div
    className="css-zzqq5x"
    data-auto-placer-item={true}
    data-auto-placer-styles="
      -ms-grid-column: 3;
      -ms-grid-row: 1;
    "
    data-has-unset-css-styles={false}
  >
    <p>
      child three
    </p>
  </div>
  <div
    className="css-bh2j43"
    data-auto-placer-item={true}
    data-auto-placer-styles="
      -ms-grid-column: 4;
      -ms-grid-row: 1;
    "
    data-has-unset-css-styles={false}
  >
    <h3>
      child four
    </h3>
  </div>
  <div
    className="css-1bmyan6"
    data-auto-placer-item={true}
    data-auto-placer-styles="
      -ms-grid-column: 1;
      -ms-grid-row: 2;
    "
    data-has-unset-css-styles={false}
  >
    <div>
      child five
    </div>
  </div>
</div>
`;

exports[`react component wrapper 4`] = `
<div
  className="css-158rfps-AutoPlace"
  colLength={1}
  data-auto-placer-container={true}
  data-children={5}
  data-col-length={1}
>
  <div
    className="css-1tdfgk4"
    data-auto-placer-item={true}
    data-auto-placer-styles="
      -ms-grid-column: 1;
      -ms-grid-row: 1;
    "
    data-has-unset-css-styles={false}
  >
    <p>
      child one
    </p>
  </div>
  <div
    className="css-1bmyan6"
    data-auto-placer-item={true}
    data-auto-placer-styles="
      -ms-grid-column: 1;
      -ms-grid-row: 2;
    "
    data-has-unset-css-styles={false}
  >
    <h2>
      child two
    </h2>
  </div>
  <div
    className="css-1em999r"
    data-auto-placer-item={true}
    data-auto-placer-styles="
      -ms-grid-column: 1;
      -ms-grid-row: 3;
    "
    data-has-unset-css-styles={false}
  >
    <p>
      child three
    </p>
  </div>
  <div
    className="css-13j0lfc"
    data-auto-placer-item={true}
    data-auto-placer-styles="
      -ms-grid-column: 1;
      -ms-grid-row: 4;
    "
    data-has-unset-css-styles={false}
  >
    <h3>
      child four
    </h3>
  </div>
  <div
    className="css-1g6amma"
    data-auto-placer-item={true}
    data-auto-placer-styles="
      -ms-grid-column: 1;
      -ms-grid-row: 5;
    "
    data-has-unset-css-styles={false}
  >
    <div>
      child five
    </div>
  </div>
</div>
`;

exports[`react component wrapper 5`] = `
<div
  className="css-mjwwwb-AutoPlace"
  colLength={5}
  data-auto-placer-container={true}
  data-children={9}
  data-col-length={5}
>
  <div
    className="css-1tdfgk4"
    data-auto-placer-item={true}
    data-auto-placer-styles="
      -ms-grid-column: 1;
      -ms-grid-row: 1;
    "
    data-has-unset-css-styles={false}
  >
    <p>
      Test grandChildren 1
    </p>
  </div>
  <div
    className="css-1pi500h"
    data-auto-placer-item={true}
    data-auto-placer-styles="
      -ms-grid-column: 2;
      -ms-grid-row: 1;
    "
    data-has-unset-css-styles={false}
  >
    <p>
      Test grandChildren 1
    </p>
  </div>
  <div
    className="css-zzqq5x"
    data-auto-placer-item={true}
    data-auto-placer-styles="
      -ms-grid-column: 3;
      -ms-grid-row: 1;
    "
    data-has-unset-css-styles={false}
  >
    <p>
      Test grandChildren 1
    </p>
  </div>
  <div
    className="css-bh2j43"
    data-auto-placer-item={true}
    data-auto-placer-styles="
      -ms-grid-column: 4;
      -ms-grid-row: 1;
    "
    data-has-unset-css-styles={false}
  >
    <p>
      Test grandChildren 1
    </p>
  </div>
  <div
    className="css-17v5j8a"
    data-auto-placer-item={true}
    data-auto-placer-styles="
      -ms-grid-column: 5;
      -ms-grid-row: 1;
    "
    data-has-unset-css-styles={false}
  >
    <p>
      Test grandChildren 1
    </p>
  </div>
  <div
    className="css-1bmyan6"
    data-auto-placer-item={true}
    data-auto-placer-styles="
      -ms-grid-column: 1;
      -ms-grid-row: 2;
    "
    data-has-unset-css-styles={false}
  >
    <p>
      Test grandChildren 2
    </p>
  </div>
  <div
    className="css-o4ag6j"
    data-auto-placer-item={true}
    data-auto-placer-styles="
      -ms-grid-column: 2;
      -ms-grid-row: 2;
    "
    data-has-unset-css-styles={false}
  >
    <p>
      Test grandChildren 2
    </p>
  </div>
  <div
    className="css-1ec5bbe"
    data-auto-placer-item={true}
    data-auto-placer-styles="
      -ms-grid-column: 3;
      -ms-grid-row: 2;
    "
    data-has-unset-css-styles={false}
  >
    <p>
      Test grandChildren 2
    </p>
  </div>
  <div
    className="css-153fhk4"
    data-auto-placer-item={true}
    data-auto-placer-styles="
      -ms-grid-column: 4;
      -ms-grid-row: 2;
    "
    data-has-unset-css-styles={false}
  >
    <p>
      Test grandChildren 2
    </p>
  </div>
</div>
`;

exports[`react component wrapper 6`] = `
<div
  className="css-1c90ywa-AutoPlace"
  data-auto-placer-container={true}
  data-children={0}
  data-col-length={0}
/>
`;

exports[`react component wrapper 7`] = `
<div
  className="css-158rfps-AutoPlace"
  data-auto-placer-container={true}
  data-children={1}
  data-col-length={1}
>
  <div
    className="css-1tdfgk4"
    data-auto-placer-item={true}
    data-auto-placer-styles="
      -ms-grid-column: 1;
      -ms-grid-row: 1;
    "
    data-has-unset-css-styles={false}
  >
    <h1>
      Single
    </h1>
  </div>
</div>
`;
